### ISSUE TITLE
[Feature]: Call lvim.builtin.which_key.on_config_done after all core logic plugins have been loaded

### ISSUE THREAD

**Issue by nahojer** [LunarVim/LunarVim#1422]
### Feature motivation

You have this very nice `on_config_done` function when customizing which-key. However, it serves no purpose when trying to configure your own LSP mappings. (Also, I want to remove the ["\<leader\>gg"] mapping because I don't use Lazygit and I find that it bloats my which-key "popup.") Can I suggest moving below code from the setup function of the which-key config (`lua/core/which-key.lua`) to `lua/core/builtins/init.lua`?

```lua
if lvim.builtin.which_key.on_config_done then
    lvim.builtin.which_key.on_config_done(which_key)
end
```

`lua/core/builtins/init.lua` could then look something like this:

```lua
local M = {}

local builtins = {
  "keymappings",
  "core.which-key",
  "core.gitsigns",
  "core.compe",
  "core.dashboard",
  "core.dap",
  "core.terminal",
  "core.telescope",
  "core.treesitter",
  "core.nvimtree",
  "core.project",
  "core.bufferline",
  "core.autopairs",
  "core.comment",
  "core.lspinstall",
  "core.lualine",
}

function M.config(config)
  for _, builtin_path in ipairs(builtins) do
    local builtin = require(builtin_path)
    builtin.config(config)
  end

  if lvim.builtin.which_key.on_config_done then
      lvim.builtin.which_key.on_config_done(which_key)
  end
end

return M
```

What do you think? Am I missing something that prevents us from doing this?

### Describe the expected solution

Want to be able to override all of the keymappings.

**Comment by nahojer** [comment]
On top of this, maybe it's also possible to switch the order in how you call `lvim.lsp.on_attach_callback`? So instead of

```go
function M.common_on_attach(client, bufnr)
  if lvim.lsp.on_attach_callback then
    lvim.lsp.on_attach_callback(client, bufnr)
    Log:debug "Called lsp.on_init_callback"
  end
  lsp_highlight_document(client)
  add_lsp_buffer_keybindings(bufnr)
  require("lsp.null-ls").setup(vim.bo.filetype)
end
```

One could do

```go
function M.common_on_attach(client, bufnr)
  lsp_highlight_document(client)
  add_lsp_buffer_keybindings(bufnr)
  require("lsp.null-ls").setup(vim.bo.filetype)
  if lvim.lsp.on_attach_callback then
    lvim.lsp.on_attach_callback(client, bufnr)
    Log:debug "Called lsp.on_init_callback"
  end
end
```
?

**Comment by abzcoding** [comment]
cc: @kylo252

**Comment by kylo252** [comment]
> Can I suggest moving below code from the setup function of the which-key config (`lua/core/which-key.lua`) to `lua/core/builtins/init.lua`?

That defeats the purpose of the function in the first place.

Can you try to be more specific about your requirements? because `<leader>gg` is a corner case. For example, I'm completely overriding `<leader>f` in my personal config.

**Comment by rebuilt** [comment]
If you just want to remove the terminal binding you can do the following
```lua
    lvim.builtin.terminal.execs = {{}} 
```

To change the keymapping to something else
```lua
  lvim.builtin.terminal.execs = {{"lazygit", "tg", "Lazygit"}}
```

If you don't want to use the terminal plugin at all
```lua
lvim.builtin.terminal.active = false
```
Then run `:PackerSync`

**Comment by jay-aye-see-kay** [comment]
I ran into more or less the same issue trying to map `gd` to  `Telescope lsp_definitions` and couldn't do it with `lsp.on_attach_callback` because it's called before `add_lsp_buffer_keybindings`. If it's possible to put `lsp.on_attach_callback` at the end of `M.common_on_attach` like @nahojer suggested that could be a good solution.

But for anyone else with this problem I found a work around using `vim.defer_fn` with a timeout of 0.

```lua
lvim.lsp.on_attach_callback = function(_, bufnr)
	vim.defer_fn(function()
		local keys = {
			["gd"] = { "<cmd>Telescope lsp_definitions<CR>", "Goto Definition" },
		}
		which_key.register(keys, { mode = "n", buffer = bufnr })
	end, 0)
end
```
